#
# madwifi-0.9.2 suppress_plaintext.diff
#
# This patch disables sending packets plaintext when F_PRIVACY is set, but
# there is no default transmit key. It improves the security, because it
# suppresses sending any data packets before a secure connection is established
# via WPA/RSN.
#
# Closes Lintrack ticket #103.
#
# 2006-09-18, Michal Wrobel <xmxwx@asn.pl>
#
--- madwifi/net80211/ieee80211_output.c
+++ madwifi/net80211/ieee80211_output.c
@@ -820,6 +820,7 @@
 				"no default transmit key (%s) deftxkey %u",
 				__func__, vap->iv_def_txkey);
 			vap->iv_stats.is_tx_nodefkey++;
+			goto bad;
 		}
 	} else
 		key = NULL;		
